README.md

```diff
@@ -1,12 +1,213 @@
-which contains an Amazon SQS queue that is subscribed to an Amazon SNS topic.
+This library enables you intercept and mock HTTP call made by an AWS Lambda. It's useful for integration testing.
 
-The construct defines an interface (`LambdaHttpInterceptorProps`) to configure the visibility timeout of the queue.
+Performing integration tests on deployed resources allows testing applications in the best prod alike environment. Thanks to the interceptor, lambda can be tested without interacting with costly external endpoints or non preprod external endpoints for example.
 
-## Useful commands
+## Getting started
 
-- `npm run build` compile typescript to js
-- `npm run watch` watch for changes and compile
-- `npm run test` perform the jest unit tests
+This library is made of a CDK Construct to instantiate in your stack and a sdk to tool the integration tests.
+
+The HttpInterceptor construct needs to be instantiated in the stack, or inside another Construct.
+
+```ts
+import { Stack, StackProps } from "aws-cdk-lib";
+
+import { Construct } from "constructs";
+import { HttpInterceptor, applyHttpInterceptor } from "http-lambda-interceptor";
+import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
+import { Runtime } from "aws-cdk-lib/aws-lambda";
+
+export class MyStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const interceptor = new HttpInterceptor(this, "HttpInterceptor");
+
+    const myLambdaFunctionThatMakesExternalCalls = new NodejsFunction(
+      this,
+      "MakeExternalCalls",
+      {
+        runtime: Runtime.NODEJS_18_X,
+        handler: "index.handler",
+        entry: './handler.ts',
+        environment: {
+          NODE_OPTIONS: "--enable-source-maps",
+        },
+      },
+    );
+
+    applyHttpInterceptor(myLambdaFunctionThatDoesExternalCalls, interceptor);
+  }
+}
+```
+
+After Deploying that part, everything is setup on the stack to perform integration tests.
+
+```typescript
+import fetch from "node-fetch";
+import { expect, describe, it } from "vitest";
+
+process.env.HTTP_INTERCEPTOR_TABLE_NAME = '<table-name-from-construct>'
+
+import { setupLambdaHttpInterceptorConfig } from "http-lambda-interceptor";
+
+import { triggerMyLambdaFunctionThatMakesExternalCalls } from './utils';
+
+describe("hello function", () => {
+  it("returns a 200", async () => {
+    await setupLambdaHttpInterceptorConfig({
+      lambdaName: '<myLambdaFunctionThatMakesExternalCalls-name>',
+      mockConfigs: [
+        {
+          url: "https://external-api.com/*",
+          method: "GET",
+          response: {
+            status: 404,
+            body: JSON.stringify({
+              errorMessage: "Not found",
+            }),
+          },
+        },
+      ],
+    });
+    const response = await triggerMyLambdaFunctionThatMakesExternalCalls();
+    expect(response.status).toBe(200);
+  });
+});
+```
+
```

Comment:
add 

To resolve `<table-name-from-construct>` and `<myLambdaFunctionThatMakesExternalCalls-name>` easily, we recommend the usage of [@swarmion/integration-tests](https://www.swarmion.dev/docs/how-to-guides/use-integration-tests/)